crates/services/src/service.rs

```diff
@@ -83,6 +83,47 @@ pub trait RunnableService: Send {
     ) -> anyhow::Result<Self::Task>;
 }
 
+/// The result of a single iteration of the service task
+pub enum TaskRunResult {
+    /// Request the task to be run again
+    Continue,
+    /// Request the task to be abandoned
+    Stop,
+    /// Request the task to be run again, but report an error
+    ErrorContinue(anyhow::Error),
+}
+
+impl TaskRunResult {
+    /// Creates a `TaskRunResult` from a `Result` where `Ok` means `Continue` and any error is reported
+    pub fn continue_if_ok<T, E: Into<anyhow::Error>>(res: Result<T, E>) -> TaskRunResult {
```

Comment:
`continue_if_ok` is a bad name. I thought it returns `Continue` only If it is `Ok`.

What do you think about `always_continue`?

Reply:
> I thought it returns Continue only If it is Ok.

Yeah. It returns `continue_` only `if_` it is `ok`. I doesn't `always_` return `continue`... I'm confused why you want me to change it.

Reply:
Maybe just `continue` then?=) For me, the main point here is that we should still continue the task. What happens with error is not important for me as a reader of the code.

Reply:
Yeah. I think you're right. I think that's what the caller is interested in. I'll go with one of those.

Reply:
I feel like TaskRunResult is the wrong name then. It's almost a state machine but not quite. But TaskState or TaskStateTransition might be a better name...